tests/test_dataclasslike.py

```diff
@@ -0,0 +1,68 @@
+import types
+import unittest
+from dataclasses import dataclass, is_dataclass
+from unittest import skipUnless
+
+from useful_types.experimental import DataclassLike
+
+class DataclassLikeTests(unittest.TestCase):
+    def test_basics(self):
+        # As an abstract base class for all dataclasses,
+        # it makes sense for DataclassLike to also be considered a dataclass
+        self.assertTrue(is_dataclass(DataclassLike))
+        self.assertTrue(issubclass(DataclassLike, DataclassLike))
+
+        with self.assertRaises(TypeError):
+            DataclassLike()
+
+        with self.assertRaises(TypeError):
+            class Foo(DataclassLike):
+                pass
+    def test_isinstance_issubclass(self):
+        @dataclass
+        class Dataclass:
```

Comment:
I guess that's because of https://github.com/hauntsaninja/useful_types/blob/ef54559627c4f421f55e22f63b83ed394795bdb4/pyproject.toml#L23 ?

Reply:
Yes, `pyi = True` seemed like a better match for useful_types proper, since you get nice and compact Protocols, like in typeshed.

I don't have a strong opinion though, I'm fine with any of a) pyi = true, b) pyi = false, c) per-project black config, d) just remove black. Up to you!

Reply:
I think my preference would actually be (d), I think? Fully agreed that for the bits of this project that are more like stubs, we'll want formatting that's more like stubs. But for things like `DataclassLike` that might have a little more runtime implementation to them, we'll probably want formatting that's more like what black would normally do with a `.py` file, even in `useful_types` proper

Reply:
I would go for b). Not having autoformatting is too much work.

Reply:
> I would go for b). Not having autoformatting is too much work.

Fine by me, I'll do that